CONTRIBUTING.md

```diff
@@ -48,6 +48,14 @@ Algorithms should:
 * return all calculation results instead of printing or plotting them
 
 Algorithms in this repo should not be how-to examples for existing Python packages.  Instead, they should perform internal calculations or manipulations to convert input values into different output values.  Those calculations or manipulations can use data types, classes, or functions of existing Python packages but each algorithm in this repo should add unique value.
+#### Pre-commit plugin
+To forget about coding style, let [pre-commit](https://pre-commit.com/#installation) fix your `flake8`/`black`/`isort` issue.
+
+```bash
+pip3 install pre-commit # required only once
+pre-commit install
+pre-commit run -a # this might show you that it failed but actually it didn't
```

Comment:
This is not required.

Reply:
Line 57: `pre-commit run -a`

Reply:
When i queried at thier repo, they told me to use `pre-commit run -a`. And after some research i found that `pre-commit install` is used to install it to `.github` directory

Reply:
You can try it locally

Reply:
I am not saying this is unnecessary but rather not required.

GitHub and git are two completely different products. `.github` directory is used only on GitHub and it has nothing to do locally. I think you meant  `.git` directory.

What `pre-commit install` does is install a hook for **git** and not **GitHub** locally in the `.git` directory. Now, whenever you commit any changes **locally**, `pre-commit` will run and will do the same thing as `pre-commit run -a` but only on the files which are included in that commit.

`pre-commit run -a`: pre-commit will only run once when you input that command.
`pre-commit install`: this will install the hook permanently and then it will run automatically whenever you want to commit any changes.

I hope you understand the difference :)

Reply:
What we can do is this:
```bash
$ pre-commit install  # install the hook and forget about coding style

# or

$ pre-commit run -a  # run the hook manually every time, too much work

Reply:
I understood it, but just to confirm i had problem committing the changes when i did not run pre-commit (can se simply remove the install line and leave the run line

Reply:
Install seems to cause problems in my case

Reply:
Run helps

Reply:
Give me some details regarding the problem.

> (can we simply remove the install line and leave the run line)

That is why I'm suggesting to include both and a comment saying what it does as mentioned here: 

> What we can do is this:
> 
> ```shell
> $ pre-commit install  # install the hook and forget about coding style
> 
> # or
> 
> $ pre-commit run -a  # run the hook manually every time, too much work
> ```

Reply:
Ok so in my PiOS repository, i ran the command `pre-commit install` and after that i ran `pre-commit run -a`. That worked. After that i made some changes and got myself into the commit problem with pre-commit. I tried to commit my changes about 5 times when i found out in the git log of my ide that pre-commit was failing.

So I think pre-commit install will ensure the user can not make changes until he runs pre-commit.

Also, please try locally the above thing once. Maybe I am wrong

Reply:
This is what my workflow looks like. As you can see, pre-commit runs automatically whenver I commit any changes.
```console
$ g st
## change-api...origin/change-api
 M algorithms_keeper/log.py

(algorithms-keeper)
17:17:00: ~/projects/algorithms-keeper on change-api [!]
$ g commit -am "Fix: log color for 304 response"
Check Yaml...........................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
black....................................................................Passed
isort....................................................................Passed
flake8...................................................................Passed
mypy.....................................................................Passed
[change-api 4c9c4db] Fix: log color for 304 response
 1 file changed, 3 insertions(+), 1 deletion(-)

(algorithms-keeper)
17:17:15: ~/projects/algorithms-keeper on change-api
$ g push
```

Reply:
I use it in all the repository including TheAlgorithms/Python :)

Reply:
Are you committing changes through the IDE git interface?
Which IDE are you talking about?

Reply:
> So I think pre-commit install will ensure the user can not make changes until he runs pre-commit run.

This is not true lol, people use `pre-commit` to forget about coding style and to maintain consistency throughout the project.

Reply:
Ok then, well I had problems with pycharm ide but this repo seems to have done a fix. Also for instance i will make a separate bash block below which will show the users to fix it manually in case of any error

Reply:
Also I will be committing changes after a while (got to go)

Reply:
You have this problem: https://github.com/xcodz-dot/PiOS/blob/main/.pre-commit-config.yaml#L23 where you copy-pasted the same configuration as this repository. The problem is your directory does not contain those files. Maybe try fixing that and see what happens.

Reply:
No worries, I will make the necessary changes and merge this PR.